paimon-core/src/main/java/org/apache/paimon/io/KeyValueDataFileWriter.java

```diff
@@ -116,6 +117,7 @@ public KeyValueDataFileWriter(
         this.dataFileIndexWriter =
                 DataFileIndexWriter.create(
                         fileIO, dataFileToFileIndexPath(path), valueType, fileIndexOptions);
+        this.isExternalPath = false;
```

Comment:
Remove it, we don't need to introduce this now.

Reply:
done